src/SOS/Strike/strike.cpp

```diff
@@ -16713,6 +16713,23 @@ DECLARE_API(SetSymbolServer)
         DisableSymbolStore();
     }
 
+    if (searchDirectory.data != nullptr) {
+        std::string fullPath;
+        if (!GetAbsolutePath(searchDirectory.data, fullPath))
+        {
+            ExtErr("Invalid search directory %s\n", fullPath.c_str());
+            return E_FAIL;
+        }
+
+        size_t length = fullPath.length() + 1;
```

Comment:
Why are you allocating and copying the searchDirectory string when you should be able to do searchDirectory.data = fullPath.c_str() as long as you move the "std::string fullPath" to the outer scope.

Reply:
```
std::string fullPath;
if (searchDirectory.data != nullptr) 
{
    if (!GetAbsolutePath(searchDirectory.data, fullPath))
    {
         ExtErr("Invalid search directory %s\n", fullPath.c_str());
         return E_FAIL;
     }
     searchDirectory.data = fullPath.c_str();
}
```

Reply:
I tried that first.

> as you move the "std::string fullPath" to the outer scope.

and this is why I was getting sigsegv. Will retry with your way. :)

Reply:
Never mind, I forgot that StringHolder delete's the string in the destructor.

Reply:
Or you could just passing `!fullPath.c_str() ?  fullPath.c_str() : nullptr` to InitializeSymbolStore() and remove the `searchDirectory.data = fullPath.c_str();`

Reply:
Ah yup, that is the real reason of sigsegv (still happening after hoisting out fullPath from current scrope). Casting `const char*` to `char*`, and letting destructor call delete[] choked lldb. :)

Reply:
Just seeing your last message. (GitHub lag or code comments don't load asynchronously)
Will update.